silk-react-components/src/HierarchicalMapping/Components/SuggestionsList.jsx

```diff
@@ -1,183 +1,147 @@
     <div className="ecc-silk-mapping__ruleslist ecc-silk-mapping__suggestionlist">
         <Card fixedActions>{props.children}</Card>
     </div>
 );
 
-const SuggestionsList = React.createClass({
-    mixins: [UseMessageBus],
-    defaultCheckValue: false,
-    // define property types
-    // FIXME: check propTypes
-    propTypes: {
-        targetClassUris: React.PropTypes.array,
-    },
-    getInitialState() {
-        return {
-            data: undefined,
-            error: false,
-            showDefaultProperties: true,
-            rawData: undefined,
-            askForDiscard: false,
-            checked: this.defaultCheckValue,
-            matchFromDataset: true,
-            warnings: [],
-        };
-    },
-    onChecked(v) {
+class SuggestionsList extends React.Component {
+    static propTypes = {
+        targetClassUris: PropTypes.array,
+        onAskDiscardChanges: PropTypes.func
+    };
+    
+    state = {
+        data: undefined,
+        error: false,
+        showDefaultProperties: true,
+        rawData: undefined,
+        checked: this.defaultCheckValue,
+        matchFromDataset: true,
+        warnings: [],
+    };
+    
+    count = 0;
+    
+    onChecked = (v) => {
         const data = this.state.data;
         const index = _.findIndex(data, d => d.id === v.id);
         data[index].checked = !data[index].checked;
         this.setState({data});
-    },
+    }
+    
     loadData() {
         this.setState({
             loading: true,
         });
-        hierarchicalMappingChannel
-            .request({
-                topic: 'rule.suggestions',
-                data: {
-                    targetClassUris: this.props.targetClassUris,
-                    ruleId: this.props.ruleId,
-                    matchFromDataset: this.state.matchFromDataset,
-                },
-            })
-            .subscribe(
-                response => {
-                    const rawData = _.map(response.suggestions, v => ({
-                        ...v,
-                        checked: this.defaultCheckValue,
-                        type: v.type || SUGGESTION_TYPES[0],
-                    }));
-                    this.setState({
-                        warnings: response.warnings.filter(w => !_.isEmpty(w)),
-                        loading: false,
-                        rawData,
-                        data: this.state.showDefaultProperties
-                            ? rawData
-                            : rawData.filter(v => !!v.targetProperty),
-                    });
-                },
-                err => {
-                    this.setState({loading: false, error: [{error: err}]});
-                }
-            );
-    },
-    onTypeChanged(v) {
+        getSuggestionsAsync({
+            targetClassUris: this.props.targetClassUris,
+            ruleId: this.props.ruleId,
+            matchFromDataset: this.state.matchFromDataset,
+        }).subscribe(
+            response => {
```

Comment:
When `this.defaultCheckValue` is always `undefined` then `rawData` and `warnings` can be moved to data transformation of `getSuggestionsAsync()` and directly returned instead of `response`.